mitodl/mit-open
# updated header

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5305

### Description (What does it do?)
This PR implements the new site header as shown in the issue above and the screenshots below.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/8881a417-6dda-466e-8d50-f20608df219d)
![image](https://github.com/user-attachments/assets/491b6589-0c29-4714-88f7-e617c1e35d11)
![image](https://github.com/user-attachments/assets/e6c9cb2c-3229-4ed8-9594-8fc6891d3a7e)
![image](https://github.com/user-attachments/assets/1d92ae1a-0ebb-4db9-b6dc-8df61fb61845)

### How can this be tested?
 - Spin up `mit-learn` on this branch
 - Load the home page at https://localhost:8062/
 - Compare the new header to the Figma designs in the above issue both while logged in and logged out and make sure everything shows up as it should
